example/main.dart

```diff
@@ -95,4 +95,40 @@ void main() async {
           postResposne.data);
     });
   });
+
+  group('Raising the custrom Error onRequest', () {
+    const path = 'https://example.com';
+    test('Test that throws raises custom exception', () async {
+      final dio = Dio();
+      final dioAdapter = DioAdapter();
+
+      dio.httpClientAdapter = dioAdapter;
+      final type = DioErrorType.RESPONSE;
+      final response = Response(statusCode: 500);
+      final error = 'Some beautiful error';
```

Comment:
I think `type`, `response` and `error` ought to be const instead of final.

Reply:
Response can't be const, it is constroctor of DIo itself and is not initialized with const values, other two can be const.